### DISCUSSION TITLE
AzuraCast Installation in Unraid system

### DISCUSSION THREAD

**Discussion by toinopt** [AzuraCast/AzuraCast#6151] [Q&A] [answered]
Hello, just found this software and it seems interesting enough that I would to try it but i can't figure out the best way to install it.

Unraid has docker support either by comunnity made templates or by using a normal docker image and then we manually add the necessary envs to the template or it tries to automatically find the correct ones.

In this case i tried the automatic option on with this https://hub.docker.com/r/azuracast/azuracast but it tried maping 1000 ports from 8000 to 9000 plus a couple of extra ones for redis and mysql and my question would be if I can install it this way but just pass the required ports for example 8000 and 9000 plus another one for the UI or would i need to add extra envs?

If i can install it this way what are the necessary envs i need to have?

The main reason not to do the recommended docker install it that i would loose the automatic updates and it would take a lot more time to update and do maintenance.

Thanks.

**Comment by BusterNeece** [comment]
We pre-populate our Docker image build with "sensible defaults" for environment variables, so you should be able to just run the image as-is.

I'd recommend clearing ports 80, 443, 2022 and then ports 8xx0, 8xx5 and 8xx6 in ascending order, like:

```
      - '8000:8000'
      - '8005:8005'
      - '8006:8006'
      - '8010:8010'
      - '8015:8015'
      - '8016:8016'
      - '8020:8020'
      - '8025:8025'
      - '8026:8026'
      - '8030:8030'
      - '8035:8035'
      - '8036:8036'
      - '8040:8040'
      - '8045:8045'
      - '8046:8046'
```

The Docker userland proxy can take up a ton of memory if you just map all of the ports from 8000-9000, which is why we specifically only map the ones that are expected to ever need to be connected to remotely.

If you enable the "Use Web Proxy" setting and don't have DJs connecting inbound, then you don't need _any_ of the ports in that range opened and can instead just use 80/443.

**Reply by toinopt** [reply]
Since I'm using NPM for the reverse proxy and I believe I have that option active, that means I can remove the couple of ports in my docker compose and now that I'm thinking about it, it makes sense not needing them since they are not used.

**Comment by toinopt** [comment, marked as answer]
If anyone ends up finding this in the future this is the docker compose i used in Unraid.

```#
# AzuraCast Docker Compose Configuration File
#
# When updating, you will be prompted to replace this file with a new
# version; you should do this whenever possible to take advantage of
# new updates.
#
# If you need to customize this file, you can create a new file named:
#   docker-compose.override.yml
# with any changes you need to make.
#

services:
  web:
    container_name: azuracast
    image: "ghcr.io/azuracast/azuracast:${AZURACAST_VERSION:-latest}"
    labels:
      - "com.centurylinklabs.watchtower.scope=azuracast"
    # Want to customize the HTTP/S ports? Follow the instructions here:
    # https://docs.azuracast.com/en/administration/docker#using-non-standard-ports
    ports:
      - '${AZURACAST_HTTP_PORT:-8147}:80'
      - '${AZURACAST_HTTPS_PORT:-8148}:443'
      - '${AZURACAST_SFTP_PORT:-8149}:2022'
      # This default mapping is the outgoing and incoming ports for the first 50 stations.
      # You can override this port mapping in your own docker-compose.override.yml file.
      # For instructions, see:
      # https://docs.azuracast.com/en/administration/docker#expanding-the-station-port-range
      - '8150:8150'
      - '8151:8151'
      - '8152:8152'
      - '8153:8153'
      - '8154:8154'
    networks:
     - dockerproxy
    env_file: .env
    environment:
      LANG: ${LANG:-en_US.UTF-8}
      AZURACAST_DC_REVISION: 14
      AZURACAST_VERSION: ${AZURACAST_VERSION:-latest}
      AZURACAST_SFTP_PORT: ${AZURACAST_SFTP_PORT:-2022}
      NGINX_TIMEOUT: ${NGINX_TIMEOUT:-1800}
      LETSENCRYPT_HOST: ${LETSENCRYPT_HOST:-}
      LETSENCRYPT_EMAIL: ${LETSENCRYPT_EMAIL:-}
      PUID: ${AZURACAST_PUID:-1000}
      PGID: ${AZURACAST_PGID:-1000}
    volumes:
      - www_uploads:/var/azuracast/uploads
      - station_data:/var/azuracast/stations
      - shoutcast2_install:/var/azuracast/servers/shoutcast2
      - stereo_tool_install:/var/azuracast/servers/stereo_tool
      - geolite_install:/var/azuracast/geoip
      - sftpgo_data:/var/azuracast/sftpgo/persist
      - backups:/var/azuracast/backups
      - acme:/var/azuracast/acme
      - db_data:/var/lib/mysql
    restart: unless-stopped
    ulimits:
      nofile:
        soft: 65536
        hard: 65536
    logging:
      options:
        max-size: "1m"
        max-file: "5"

  updater:
    container_name: azuracast_updater
    image: ghcr.io/azuracast/updater:latest
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    logging:
      options:
        max-size: "1m"
        max-file: "5"

volumes:
    db_data: {  }
    acme: {  }
    shoutcast2_install: {  }
    stereo_tool_install: {  }
    geolite_install: {  }
    sftpgo_data: {  }
    station_data: {  }
    www_uploads: {  }
    backups: {  }
    www_vendor: {  }

networks:
    dockerproxy:
        external: true
    ```
